src/Loop.php

```diff
@@ -59,155 +59,162 @@ public static function stop()
     /**
      * Defer the execution of a callback.
      *
-     * @param callable $callback The callback to defer.
+     * @param callable(mixed $data, string $watcherIdentifier) $callback The callback to defer.
+     * @param mixed $data Arbitrary data given to the callback function as the $data parameter.
      *
-     * @return string An identifier that can be used to cancel, enable or disable the event.
+     * @return string An identifier that can be used to cancel, enable or disable the watcher.
      */
-    public static function defer(callable $callback)
+    public static function defer(callable $callback, $data = null)
     {
-        return self::get()->defer($callback);
+        return self::get()->defer($callback, $data);
     }
 
     /**
      * Delay the execution of a callback. The time delay is approximate and accuracy is not guaranteed.
      *
-     * @param callable $callback The callback to delay.
+     * @param callable(mixed $data, string $watcherIdentifier) $callback The callback to delay.
```

Comment:
`$data` is optional for the method creating the watcher, but the argument would always be provided to the callback, even if that value is `null`. I felt the data argument being first would make sense in more use-cases, since most watcher callbacks won't care about their own identifier.